docs/FormatsAndConventions.md

```diff
@@ -27,20 +27,96 @@ Commit messages for non-trivial commits should have a _**body**_ giving details
  
 1. Separate subject from body with a blank line
 1. Wrap the body at 72 characters
-1. Use the body to explain what and why vs. how
-
-Here is an example (adapted from [this commit](https://github.com/CS2103AUG2016-T11-C4/main/pull/2/commits/5c5b12d1a75c4a73a8330bfd05e406233694ffa3)):
- 
- ```
- Group overloaded methods together
- 
- To make it easier to see the different ways a method can be called, 
- we should group them together.
- 
- Note that this is not required strictly by the coding standard.
- ```
+1. Use the body to explain WHAT the commit is about and WHY it was done that way. The reader can refer to the diff to 
+   understand HOW the change was done.
   
-Refer to the article _[How to Write a Git Commit Message](http://chris.beams.io/posts/git-commit/)_ for a more detailed explanation.
+   * Give an explanation for the change(s) that is detailed enough so that the reader can judge if it is a
+     good thing to do, without reading the actual diff to determine how well the code does what the explanation 
+     promises to do.<br>
+     If your description starts to get too long, that's a sign that you probably need to split up 
+     your commit to finer grained pieces. [adapted from: [git project][git-commit-guide]]
+   * Minimize in the commit message information that can be included in the code as comments. 
+
+Stylistic recommendations:
+
+* Follow this flow:
+
+   ```
+   {current situation} -- use present tense
+   
+   {why it needs to change}
+   
+   {what is being done about it} -- use imperative mood
+   
+   {why it is done that way}
+   
+   {any other relevant info}
+   ```
+   
+* Use blank lines to separate paragraphs. 
+* Avoid terms such as 'currently', 'originally' when describing the current situation. They are implied.
+* The word `Let's` can be used to indicate the beginning of the section that describes the change done in the commit.
+
+Here is an example commit message for a code quality refactoring:
+
+```
+Person attributes classes: extract parent class
```

Comment:
@damithc I think the commit subject should at least have a mention of *what* class is extracted, namely `PersonAttribute`. Otherwise, it is too vague.

On a separate note I'm guessing that the 50-character limit recommendation came from Linux kernel development, where they use C and their naming is a lot terser than what we expect in a Java project, so they have no trouble fitting their subjects within 50 characters. Even though we use longer names like `PersonAttribute`, we shouldn't be dropping these important names from the commit subject just to make it shorter.